firestore/ios/firestore-snippets-cpp/AppDelegate.mm

```diff
@@ -17,17 +17,23 @@
 #import "AppDelegate.h"
 #import <Firebase/Firebase.h>
 #include "firebase/app.h"
+#include "firebase/firestore.h"
 
 @interface AppDelegate ()
 
 @end
 
 @implementation AppDelegate
 
-
 - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
   [FIRApp configure];
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunused-variable"
+  // [START get_firestore_instance]
   firebase::App::Create();
+  auto db = firebase::firestore::Firestore::GetInstance();
+  // [END get_firestore_instance]
+#pragma clang diagnostic pop
```

Comment:
Optional: a simple workaround to silence the warning is to add a statement that only contains the name of the variable (it is a valid expression that results in the value of the variable):
```cpp
int foo = 0;
foo;
```
However, this runs into another warning saying that the expression result is unused. A common way to get rid of it is to cast the result of the expression to `void` (using a C-style cast for brevity):
```cpp
int foo = 0;
(void)foo;
```

It's ugly but shorter; as long as the `(void)db;` expression is not included in the snippet, it should be okay.